src/app/shared/paging/episode-page.component.ts

```diff
@@ -20,24 +31,45 @@ export class EpisodePageComponent implements OnChanges {
   @Input() totalPages;
   @Output() pageChange = new EventEmitter<number>();
   pages: number[];
-  showNumPages = 10;
+  showNumPages = 5;
   pagesBegin: number;
   pagesEnd: number;
+  lastPage: number;
 
   ngOnChanges() {
+    this.lastPage = Math.floor(this.totalPages);
+
+    if (this.totalPages <= this.showNumPages) {
+      this.pagesBegin = 1;
+      this.pagesEnd = this.lastPage;
+    }
+    else {
+      var halfWindow = Math.floor(this.showNumPages / 2);
+      if (this.currentPage <= (halfWindow + 1)) {
+        this.pagesBegin = 1;
+        this.pagesEnd = this.showNumPages;
+      }
+      else if ((this.currentPage + halfWindow) >= this.lastPage) {
+        this.pagesBegin = this.lastPage - this.showNumPages + 1;
+        this.pagesEnd = this.lastPage;
+      }
+      else {
+        this.pagesBegin = this.currentPage - halfWindow;
+        this.pagesEnd = this.currentPage + halfWindow;
+      }
+    }
+
     this.pages = [];
-    for (let i = 1; i <= this.totalPages; i++) {
+    for (let i = this.pagesBegin; i <= this.pagesEnd; i++) {
```

Comment:
instead of an array of all pages and then slicing in the html section, create an array of only the visible page numbers and a simple for loop in the html.